modAionImpl/src/org/aion/zero/impl/db/AionContractDetailsImpl.java

```diff
@@ -181,13 +216,71 @@ public void decode(byte[] rlpCode, boolean fastCheck) {
             externalStorage = true;
             storageTrie.getCache().setDB(getExternalStorageDataSource());
         }
+    }
+
+    /**
+     * Decodes the new version of encoding which is a list of 6 elements, specifically:<br>
+     * { 0:address, 1:vmType, 2:isExternalStorage, 3:storageRoot, 4:storage, 5:code }
```

Comment:
the rlp encode/decode order can be the original order plus the vmType. Then we can make sure the logic will be more neat and safe.